core/iwasm/common/wasm_native.c

```diff
@@ -394,6 +407,155 @@ wasm_native_unregister_natives(const char *module_name,
     return false;
 }
 
+static uint32
+context_key_to_idx(void *key)
+{
```

Comment:
And not very sure what is the difference from wasm_runtime_set_custom_data, which also binds a custom pointer to the module instance? Is it because that `wasm_runtime_module_instance_{set|get}_context` can set/get context with key while `wasm_runtime_{set|get}_custom_data` can not?

Reply:
> Is it because that wasm_runtime_module_instance_{set|get}_context can set/get context with key while wasm_runtime_{set|get}_custom_data can not?

right.
i want something which can be used by multiple subsystems.

besides that, i want it to have a cleanup callback on instance destory.